.github/workflows/bench-core.yml

```diff
@@ -0,0 +1,96 @@
+name: Benchmark Core
+on:
+  push:
+    branches: [ master ]
+  workflow_dispatch:
+
+jobs:
+  bench_core_geth:
+    name: Benchmark core-geth
+    runs-on: ubuntu-latest
+    timeout-minutes: 360
+    steps:
+      - name: Set up Go 1.x
+        id: go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ^1.16
+
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+
+      - name: Benchmark etclabscore/core-geth
+        id: bench
+        run: |
+          go test -short ./core -count 1 -p 1 -timeout 60m -run NONE -bench=. -v |& tee core-geth.txt
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: core-geth
+          path: ./core-geth.txt
+
+  bench_go_ethereum:
+    name: Benchmark go-ethereum
+    runs-on: ubuntu-latest
+    timeout-minutes: 360
+    steps:
+      - name: Set up Go 1.x
+        id: go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ^1.16
+
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+        with:
+          repository: 'ethereum/go-ethereum'
+          ref: 'v1.9.25'
```

Comment:
It's important to not forget to update the version here from now on. Wondering if we can fetch this version from somewhere in another step and use it here.

Reply:
Yea, this crossed my mind too. The issue with dynamic updates would be that there are no guarantees that benchmark tests will stay constant across upstream versions (although in practice, this is probably an edge case); and I wanted to maintain consistency for the test control group. My thinking was that when we start to actually use these, we _should_ expect ourselves to have the basic states of the test groups in mind when analyzing their results.

I'm not hard and fast on hardcoding it, but that was my reasoning. If you'd like to push for dynamic versioning I would be more than happy to see what you have in mind.